resources/types/subnet.go

```diff
@@ -61,17 +63,50 @@ func NewSubnet(resourceId string, subnet *resourcespb.SubnetArgs, others *resour
 	return s, nil
 }
 
-func SubnetFromState(r *Subnet, _ *output.TfState) (*resourcespb.SubnetResource, error) {
-	return &resourcespb.SubnetResource{
-		CommonParameters: &commonpb.CommonChildResourceParameters{
-			ResourceId:  r.ResourceId,
-			NeedsUpdate: false,
-		},
-		Name:             r.Args.Name,
-		CidrBlock:        r.Args.CidrBlock,
-		AvailabilityZone: r.Args.AvailabilityZone,
-		VirtualNetworkId: r.Args.VirtualNetworkId,
-	}, nil
+func SubnetFromState(r *Subnet, state *output.TfState) (*resourcespb.SubnetResource, error) {
+	if flags.DryRun {
+		return &resourcespb.SubnetResource{
+			CommonParameters: &commonpb.CommonChildResourceParameters{
+				ResourceId:  r.ResourceId,
+				NeedsUpdate: false,
+			},
+			Name:             r.Args.Name,
+			CidrBlock:        r.Args.CidrBlock,
+			AvailabilityZone: r.Args.AvailabilityZone,
+			VirtualNetworkId: r.Args.VirtualNetworkId,
+		}, nil
+	}
+	out := new(resourcespb.SubnetResource)
+	out.CommonParameters = &commonpb.CommonChildResourceParameters{
+		ResourceId:  r.ResourceId,
+		NeedsUpdate: false,
+	}
+	out.AvailabilityZone = r.Args.AvailabilityZone
+
+	id, err := resources.GetMainOutputRef(r)
+	if err != nil {
+		return nil, err
+	}
+	switch r.GetCloud() {
+	case common.AWS:
+		stateResource, err := output.GetParsed[subnet.AwsSubnet](state, id)
+		if err != nil {
+			return nil, err
+		}
+		out.Name = stateResource.AwsResource.Tags["Name"]
+		out.CidrBlock = stateResource.CidrBlock
+		out.VirtualNetworkId = stateResource.VpcId
```

Comment:
Here you should use r.Args.GetVirtualNetworkId()

This ID represents the ID generated by Multy, not the one generated by each cloud provider. We don't actually track the ones generaed by each cloud provider (but will at some point)

Reply:
Thanks for the review and explaining the changes. I'll make the changes.